manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -142,6 +142,7 @@ func (na *cnmNetworkAllocator) Allocate(n *api.Network) error {
 		n.DriverState = &api.Driver{
 			Name: d.name,
 		}
+		n.IPAM = &api.IPAMOptions{Driver: &api.Driver{}}
```

Comment:
Can you add a comment about why this is necessary?

Also, I'm not sure this should be done at allocation time. That leaves a window where it could crash an earlier version. Shouldn't it be done at the time the network is created?